common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerFactory.scala

```diff
@@ -73,6 +75,15 @@ case class ContainerPoolConfig(userMemory: ByteSize,
   // Grant more CPU to a container if it allocates more memory.
   def cpuShare(reservedMemory: ByteSize) =
     max((totalShare / (userMemory.toBytes / reservedMemory.toBytes)).toInt, 2) // The minimum allowed cpu-shares is 2
+
+  private val minContainerCpus = 0.01 // The minimum cpus allowed by docker is 0.01
+  def cpuLimit(reservedMemory: ByteSize): Option[Double] = {
+    userCpus.map(c => {
+      val containerCpus = c / (userMemory.toBytes / reservedMemory.toBytes)
+      val roundedContainerCpus = round(containerCpus * 100).toDouble / 100 // Docker only allows decimal precision of 2
+      max(roundedContainerCpus, minContainerCpus)
```

Comment:
So the expected cpu limit with the default configuration would be 0.125 core for 256MB?

Reply:
It depends on how many total cores are configured. Looks like the default userMemory is 1024MB, so with 1 total core configured the limit would be 0.25 cores for 256MB. With 2 cores configured, the limit would be 0.5 cores for 256MB. I left the total cores not configured by default though so no limit is applied unless that is set.

Reply:
Got it, the below `user-cpus` is commented.
I think the default userMemory is 2048MB.
https://github.com/apache/openwhisk/blob/master/ansible/group_vars/all#L211

So when the default `userMemory` is 2048 and the `user-cpus` is 1, then it would be 0.125 core.

Reply:
Ah, I was looking at the invoker application.conf. So yeah with those settings 0.125 core is correct.